### ISSUE TITLE
Problem with subsections

### ISSUE THREAD

**Issue by jorgesca** [myint/rst2beamer#5]
Hi, I am having a problem with subsections, but I am not sure if it is because rst2beamer (because I tested the original rst2beamer (at https://github.com/rst2beamer/rst2beamer) and I get the same error.
pdflatex is giving the following error:

`sectioning_test.tex|152 error| Argument of \@gobble has an extra }.`
`sectioning_test.tex|152 error| Paragraph ended before \@gobble was complete.`
`sectioning_test.tex|152 error| Argument of \@firstofone has an extra }.`
`sectioning_test.tex|152 error| Paragraph ended before \@firstofone was complete.`
`sectioning_test.tex|152 error| Undefined control sequence.`

The body of the tex file is below. I think the problem is the label of sections and subsections occurring inside the section title, bu I can't figure out wher it is coming from. Coult it be a change in the latex writer? This was tested with the latest commit as of now.

```tex
%%% Body
\begin{document}


\section{Section 1%
  \label{section-1}%
}


\subsection{Subsection 1.1%
  \label{subsection-1-1}%
}

\begin{frame}[fragile]
\frametitle{Slide 1}

\begin{itemize}

\item These slides are in sections and subsections within the document
\begin{itemize}

\item Which means all slides must be in a section
\emph{and} subsection! (For sectioning consistency.)

\item You may use a \textquotedbl{}dummy\textquotedbl{} subsection to make a slide appear to be in a
section rather than a subsection.
\end{itemize}
\end{itemize}

\end{frame}


\subsection{Subsection 1.2%
  \label{subsection-1-2}%
}

\begin{frame}[fragile]
\frametitle{Slide 2}


A slide in a new subsection.

\end{frame}


\section{Section 2%
  \label{section-2}%
}


\subsection{Subsection 2.1%
  \label{subsection-2-1}%
}

\begin{frame}[fragile]
\frametitle{Slide 3}


A slide.

\end{frame}

\begin{frame}[fragile]
\frametitle{Slide 4}


A slide.

\end{frame}


\section{Section 3%
  \label{section-3}%
}

\begin{frame}[fragile]
\frametitle{Slide 5}


These next two slides are in a \textquotedbl{}dummy\textquotedbl{} subsection

\end{frame}

\begin{frame}[fragile]
\frametitle{Slide 6}


A slide.

\end{frame}

\end{document}
```

`

**Comment by myint** [comment]
The output of of the LaTeX writer hasn't changed since 2014.

https://github.com/myint/rst2beamer/blob/master/test/sectioning_test_expected.tex

https://github.com/myint/rst2beamer/blob/master/test.bash

**Comment by jorgesca** [comment]
You're right. I just run test.bash locally and the result is OKAY. pdflatex still complains about the generated latex code, though, so I kept scratching my head. Then I remembered I use the development version of beamer. I temporarily switched to the version shipped with texlive and all worked. I'll dig some more to find why the development version of beamer breaks rst2beamer.
Sorry for the noise, and thanks.
